File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int,int> frq;
        for (int i = 0; i < nums.size(); i++) {
            if (frq[nums[i]]) return true;
            frq[nums[i]]++;
        }
        return false;
    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cs
public class Solution {
    public int FindDuplicate(int[] nums) {
        return FindDuplicateNumber(nums);
    }
    private int FindDuplicateNumber(int[] nums) {
        for (int i = 0; i < nums.Length; i++) {
            int idx = Math.Abs(nums[i]) - 1;
            if (nums[idx] < 0) return Math.Abs(nums[i]);
            nums[idx] *= -1;
        }
        return -1;
    }
}

